ftw/file/imaging.py

```diff
@@ -0,0 +1,39 @@
+from Products.Archetypes.utils import shasattr
+from ftw.file.interfaces import IFtwFileField
+from plone.app.blob.mixins import ImageFieldMixin
+from plone.app.imaging.traverse import DefaultImageScaleHandler
+from zope.component import adapts
+from zope.interface import implementsOnly
+
+
+class FtwFileImageScaleHandler(DefaultImageScaleHandler):
+    adapts(IFtwFileField)
+
+    def getScale(self, instance, scale):
+        if self.context.get(instance).content_type.startswith('image/'):
+            return super(FtwFileImageScaleHandler, self).getScale(instance,
+                                                                  scale)
+        else:
+            return None
+
+
+class ImagingMixin(ImageFieldMixin):
+    implementsOnly()
```

Comment:
@Tschanzt could you add a comment here, explaining what the purpose / idea of `implementsOnly` and the `ImageMixin` is? It's not that obviuous :stuck_out_tongue_winking_eye: